samples/quickperf/src/main/java/com/google/cloud/jdbc/quickperf/QuickPerf.java

```
@@ -0,0 +1,497 @@
+        System.out.println("Query: " + config.getQuery());
+        System.out.println("Params: " + config.paramsToString());
+        System.out.println("Tag: " + DEFAULT_TAG);
+        if (config.getBatchSize() > 0) {
+            System.out.println("Batching Enabled (size): " + config.getBatchSize());
+        }
+        System.out.println(String.format("Start: %s End: %s", testStartTimestamp, ZonedDateTime.now()));
+        System.out.println(
+                String.format(
+                        "Finished with a total of %s runs across %s Threads.\nLatencies (ms):  p50 = %s, p95 = %s, p99 = %s, min = %s,  max = %s",
+                        config.getIterations() * config.getThreads(), config.getThreads(),
+                        calcPerc(measures, 50), calcPerc(measures, 95), calcPerc(measures, 99), getMin(measures),
+                        getMax(measures)));
+        System.out.println(BREAK_STR);
+
+    }
+
+    public QuickPerf(Config config) {
+        this.config = config;
+    }
+
+    private void runSampling() {
+        System.out.println("Running Sampling... ");
+
+        String connectionUrl = createConnectionURL(config);
+
+        try (Connection connection = DriverManager.getConnection(connectionUrl)) {
+            try (Statement statement = connection.createStatement()) {
+                boolean hasResults = statement.execute(config.getSamplingQuery());
+
+                if (hasResults == false) {
+                    System.out.println("Nothing sampled");
+                    return;
+                }
+
+                ResultSet rs = statement.getResultSet();
+                while (rs.next()) {
+                    String value = rs.getString(1);
+                    sampledValueList.add(value);
+                }
+
+                System.out.println(String.format("Finished sampling %s records", sampledValueList.size()));
+            } catch (SQLException e) {
+                e.printStackTrace();
```

I think that it is better to propagate this error to the caller, and just make the entire benchmark fail if sampling fails.

---

Changed to propagate exception to fail entire app